view/adminhtml/templates/new_shipment.phtml

```diff
@@ -1,179 +1,253 @@
+                        <?php
+                        while ($label_amount <= 10) {
+                            echo '<option value="' . $label_amount . '" >' . __($label_amount) . '</option>';
+                            $label_amount++;
+                        }
+                        ?>
                     </select>
                 </div>
-                <div class="admin__field admin__field-option">
-                    <input type="radio" class="admin__control-radio" id="mypa_package_type-mailbox" name="mypa_package_type" value="mailbox" <?=$block->getPackageType() !== 2 ?: 'checked="checked"';?>>
-                    <label class="admin__field-label" for="mypa_package_type-mailbox"><?=__('Mailbox');?></label>
+                <?php
+                try {
+                    $abstractConsignment->setPackageType($packageTypeId);
+                } catch (\Exception $e) {
+                    continue;
+                }
+                foreach ($abstractConsignment->getAllowedShipmentOptions() as $shipmentOption) {
+                    if (AbstractConsignment::SHIPMENT_OPTION_INSURANCE === $shipmentOption
+                        || ! $abstractConsignment->canHaveShipmentOption($shipmentOption)) {
+                        continue;
+                    }
+                    $isActive = ('large_format' === $shipmentOption)
+                        ? $block->getDefaultLargeFormat($shipmentOption)
+                        : $block->getDefaultOption($shipmentOption);
+                    ?>
+                    <div
+                        class="field choice admin__field admin__field-option field-mypa_signature mypa_package-toggle mypa-option-toggle"
+                        data-for_mypa_package_type="<?= $packageTypeName ?>"
+                        style="padding-left:30px;">
+                        <input
+                            id="mypa_<?= $shipmentOption ?>_<?= $carrierName ?>_<?= $packageTypeId ?>"
+                            class="admin__control-checkbox"
+                            name="mypa_<?= $shipmentOption ?>"
+                            value="1"
+                            type="checkbox"
+                            <?= $isActive
+                                ? 'checked="checked"'
+                                : '' ?>
+                        />
+                        <label
+                            class="admin__field-label"
+                            for="mypa_<?= $shipmentOption ?>_<?= $carrierName ?>_<?= $packageTypeId ?>">
+                            <span><?= $block->getNewShipmentForm()
+                                          ->getShipmentOptionsHumanMap()[$shipmentOption] ?? $shipmentOption ?></span>
+                        </label>
+                    </div>
+
+                    <?php
+                }
+
+                if (! $abstractConsignment->canHaveShipmentOption(AbstractConsignment::SHIPMENT_OPTION_INSURANCE)) {
+                    continue;
```

Comment:
Misschien is het toch nicer om hier geen continue te gebruiken maar wel een normaal if blokje, lijkt mij iets explicieter, nu werkt het alleen omdat insurance toevallig de laatste optie is